src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -242,117 +374,156 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
                         lazyLoad: false
                     });
 
-                    flvPlayer.attachMediaElement(elem);
-                    flvPlayer.load();
+                flvPlayer.attachMediaElement(elem);
+                flvPlayer.load();
 
-                    flvPlayer.play().then(resolve, reject);
-                    instance._flvPlayer = flvPlayer;
+                this.#flvPlayer = flvPlayer;
 
-                    // This is needed in setCurrentTrackElement
-                    self._currentSrc = url;
-                });
+                // This is needed in setCurrentTrackElement
+                this.#currentSrc = url;
+
+                return flvPlayer.play();
             });
         }
 
-        function setSrcWithHlsJs(instance, elem, options, url) {
-            return new Promise(function (resolve, reject) {
-                requireHlsPlayer(function () {
-                    var hls = new Hls({
+        /**
+         * @private
+         */
+        setSrcWithHlsJs(elem, options, url) {
+            return new Promise((resolve, reject) => {
+                requireHlsPlayer(() => {
+                    const hls = new Hls({
                         manifestLoadingTimeOut: 20000,
-                        xhrSetup: function(xhr, xhr_url) {
+                        xhrSetup(xhr) {
                             xhr.withCredentials = true;
                         }
                     });
                     hls.loadSource(url);
                     hls.attachMedia(elem);
 
-                    htmlMediaHelper.bindEventsToHlsPlayer(self, hls, elem, onError, resolve, reject);
+                    bindEventsToHlsPlayer(this, hls, elem, this.onError, resolve, reject);
 
-                    self._hlsPlayer = hls;
+                    this._hlsPlayer = hls;
 
                     // This is needed in setCurrentTrackElement
-                    self._currentSrc = url;
+                    this.#currentSrc = url;
                 });
             });
         }
 
-        function setCurrentSrc(elem, options) {
-            elem.removeEventListener('error', onError);
+        /**
+         * @private
+         */
+        setCurrentSrcChromecast(instance, elem, options, url) {
+            elem.autoplay = true;
+
+            const lrd = new cast.receiver.MediaManager.LoadRequestData();
+            lrd.currentTime = (options.playerStartPositionTicks || 0) / 10000000;
+            lrd.autoplay = true;
+            lrd.media = new cast.receiver.media.MediaInformation();
+
+            lrd.media.contentId = url;
+            lrd.media.contentType = options.mimeType;
+            lrd.media.streamType = cast.receiver.media.StreamType.OTHER;
+            lrd.media.customData = options;
 
```

Comment:
#1599
```suggestion
```
_I had to split suggestion - GH is seeing line deletion somewhere :man_shrugging: _

Reply:
already deleted locally